content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
Based on trying to keep the API as simple as possible from a developer's perspective, yet allowing the options to scale later, why don't we simply have:

```javascript
channel.get({ cipher: { key: key }});
```

If the dev wants to add more params they can do any of the following:

```javascript
channel.get({ cipher: { key: key, mode: 'cbc' }});
// or
var cipherParams = Cypto.getDefaultParams({key: key, mode: 'cbc'});
channel.get({ cipher: cipherParams });
```

Calling it `cipherParams` as opposed to `cipher` makes it sound like you're configuring it but not necessarily enabling it on the channel unless `encrypted: true` is included.  And we can still specify that the type expected is a `CipherParams` object.

WDYT?